jni/src/faiss_wrapper.cpp

```diff
@@ -217,10 +241,6 @@ void knn_jni::faiss_wrapper::CreateIndexFromTemplate(knn_jni::JNIUtilInterface *
     auto idVector = jniUtil->ConvertJavaIntArrayToCppIntVector(env, idsJ);
     faiss::IndexIDMap idMap =  faiss::IndexIDMap(indexWriter.get());
     idMap.add_with_ids(numVectors, inputVectors->data(), idVector.data());
-    // Releasing the vectorsAddressJ memory as that is not required once we have created the index.
-    // This is not the ideal approach, please refer this gh issue for long term solution:
-    // https://github.com/opensearch-project/k-NN/issues/1600
-    delete inputVectors;
```

Comment:
there was a reason for deallocating the memory here. Please read the gh issue: https://github.com/opensearch-project/k-NN/issues/1600 . The iterative graph build solution is only for HNSW index and not for training based indices. So we should keep this de-allocation here.

Reply:
> The iterative graph build solution is only for HNSW index and not for training based indices. So we should keep this de-allocation here.

The code path of vector transfer doesn't defer, the same applies for allocating and deallocating of memory for these input vectors.

I strongly believe that memory management shouldn't be at abrupt locations in the code base. We should be trying to keep the control of the delete with the location where it was allocated to keep it centralized and make the code readable, moreover it removes the responsibility of deallocating the memory from various places in the code

As far as the comment goes, the github issue will track the iterative approach for nmslib and other cases as well

Reply:
> I strongly believe that memory management shouldn't be at abrupt locations in the code base. 

I also agree on this. But in this case you have to release the memory here, otherwise you will go to the next steps in index writing and still holding the memory which is not required. Hence the comment was added clearly stating why we should do this.

Also this code path is for **template index** and not for iterative builds we should not change the memory release pattern otherwise you will hold more memory even when you start writing the index.

> As far as the comment goes, the github issue will track the iterative approach for nmslib and other cases as well

Yes it should but since we are not making the change for nmslib and other cases we should free up the memory in java layer. Specially in case of nmslib which increases the memory footprint when it write the index on the file.